docs/ocm-control-plane-walkthrough.md

```diff
@@ -0,0 +1,267 @@
+# Open Cluster Management and Multi-Cluster gateways
+
+
+This document will walk you through using Open Cluster Management and Kuadrant to deploy and integrate with a multi-cluster gateway. 
+
+
+## Requirements
+
+- Kind
+- AWS Route 53 (for now)
+
+## Installation and Setup
+
+- Clone this repo locally 
+- run `make local-setup MCTC_WORKLOAD_CLUSTERS_COUNT=2`
+- setup a `.controller-config.env` with the following key values
+
+```
+    # this sets up your default managed zone
+    AWS_DNS_PUBLIC_ZONE_ID=<AWS ZONE ID>
+    ZONE_ROOT_DOMAIN=some.domain.com
+```   
+
+- setup a `.aws-credentials.env` with credentials to access route 53
+
+
+### Setup OCM and Register clusters
+
+- Install OCM cli:  
+`curl -L https://raw.githubusercontent.com/open-cluster-management-io/clusteradm/main/install.sh | bash`
+- Set the hub context:
+`export CTX_HUB_CLUSTER=kind-mctc-control-plane`
+- Install OCM hub components
+`clusteradm init --wait --context ${CTX_HUB_CLUSTER}`
+
+Once installation is finished copy the command and token outputted to be used later. Add the following flag to the end of the command
+` --force-internal-endpoint-lookup `
+- create two additional terminal windows and set the kubeconfig for the spoke clusters
+
+session/window 1
+```
+kind export kubeconfig --name=mctc-workload-1 --kubeconfig=$(pwd)/local/kube/workload1.yaml && export KUBECONFIG=$(pwd)/local/kube/workload1.yaml
+```
+
+session/window 2
+
+
+```
+kind export kubeconfig --name=mctc-workload-2 --kubeconfig=$(pwd)/local/kube/workload2.yaml && export KUBECONFIG=$(pwd)/local/kube/workload2.yaml
+```
+
+- register the spoke clusters
+
+In each window where you set the kubeconfig run the command you copied earlier. The name of the cluster is up to you. I used gateway-1 and gateway-2
+```
+example
+
+clusteradm join --hub-token the_token --hub-apiserver https://127.0.0.1:49710 --wait --cluster-name gateway-1 --force-internal-endpoint-lookup
+
+clusteradm join --hub-token the_token --hub-apiserver https://127.0.0.1:49710 --wait --cluster-name gateway-2 --force-internal-endpoint-lookup
+```
+Once the gateway clusters have registered you will need to accept them in the hub back in the terminal that has the hub cluster context set 
+
+```
+clusteradm accept --clusters gateway-1
```

Comment:
when registering the spoke cluster ive named them the same as what kind names them i.e kind-mctc-workload-1 i can change that to match this